### PR TITLE
Form branch

### PR DESCRIPTION
**Pull request by soesandarwin2201**
## What does this PR do?
This PR is raised to add forms to Blog app..

## Description of the task to be completed
- [x] Create a method called current_user in your ApplicationController to make current user data available to all controllers.
- [x] It will return the first user from the database.
- [x] Don't worry about proper authentication at this time. You will add it in the upcoming projects.
- [x] Create forms to perform the following functions:
- [x] Creates a Post on behalf of the current_user (use the method that you created in your ApplicationController).
- [x] Create a comment on behalf of the current_user (use the method that you created in your ApplicationController).
- [x] Allow Users to add likes to Posts.

## How should it be tested
Clone the repository
```
git clone https://github.com/soesandarwin2201/blog_with_ruby_on_rails
```
Navigate to the folder
```
cd blog_with_ruby_on_rails
```
Checkout the branch
```
git checkoutform-branch
```
#### Thanks for reviewing my PR :blush:

### CODE REVIEW

File: app/controllers/posts_controller.rb [hunk 1]
```diff
@@ -4,8 +4,38 @@ def index
...
+      commentscounter: 0,
+      likescounter: 0
+    )
+    if @post.save
+      @post.update_user_posts_counter
+      redirect_to user_posts_path current_user.id
+    else
+      render :new
+    end
+  end
```
Comment (fobadara) [inline app/controllers/posts_controller.rb]: [REQUIRED] There are a lot of unnecessary repetitions, you have a strong params method but you are not using it. Kindly structure your create action like this instead:
```
    def create
      @post = @current_user.posts.new(post_params)

      respond_to do |format|
        if @post.save
          format.html { redirect_to @post }
        else
          format.html { render :new }
        end
      end
    end
```

File: app/controllers/comments_controller.rb [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+    @comment = Comment.new
+    respond_to do |format|
+      format.html { render :new, locals: { post: @comment } }
+    end
+  end
+
+  def create
+    @post = Post.find_by_id(params[:post_id])
+    @comment = Comment.new(strong_params)
+    @comment.post = @post
```
Comment (fobadara) [inline app/controllers/comments_controller.rb:12]: [OPTIONAL] Writing your code like below looks more professional and shorter:

```ruby
 @comment = @current_user.comments.new(params)
    @comment.post_id = params[:post_id]
```

File: app/controllers/comments_controller.rb [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+    @comment.author = current_user
+    if @comment.save
+      redirect_to user_post_path(current_user, @post)
+    else
+      render :new
+    end
```
Comment (fobadara) [inline app/controllers/comments_controller.rb:18]: [REQUIRED] Kindly redirect to the post page instead. This is better for user experience.

File: app/controllers/like_controller.rb [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+  end
+
+  def create
+    @like = Like.new
+    @like.author = current_user
+    @like.post = Post.find(params[:post_id])
+    if @like.save
+      redirect_to user_post_path(@like.post.author, @like.post)
+    else
+      render :new
```
Comment (fobadara) [inline app/controllers/like_controller.rb:9]: [REQUIRED] I believe redirecting to post will be a better user experience

File: app/controllers/like_controller.rb [hunk 5]
```diff
@@ -0,0 +1,19 @@
+class LikeController < ApplicationController
+  def new
+    @like = Like.new
+    respond_to do |format|
+      format.html { render :new, locals: { post: @like } }
+    end
+  end
```
Comment (fobadara) [inline app/controllers/like_controller.rb]: [REQUIRED] Kindly remove the new action since you won't be needing a form just to add likes

**Review by fobadara — CHANGES_REQUESTED** [review]
Hi @soesandarwin2201, 

Good job so far!
You still need to work on some issues to go to the next project, but you are almost there!

## Highlights
✅ Linters are passing
✅ Good first attempt

## Required Changes ♻️
_Check the comments under the review._
I had to include the comment here because it could not be done inline
[REQUIRED] Kindly remove the `Microverse README Template` text at the top of your README file. This will make your project look more professional.
[REQUIRED] Kindly remove the live demo section since you don't have any link to add there.

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you consider them as they can improve your code._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by BenMukebo — APPROVED** [review]
### Hi @soesandarwin2201, 👋 

# STATUS: Approved  🎉 

Great work on making the changes requested by a previous reviewer :1st_place_medal:  
 Your project is complete!  There is nothing else to say other than... it's time to merge it
Congratulations! 🎉  🎉 🎉 



### To Highlight :1st_place_medal: 
- Your Pull Request has a proper title and description. ✔️  
- ApplicationController contains #current_user method. ✔️  
- A post is created on behalf of the currently logged in user. ✔️  

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me on( @BenMukebo ) in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**